src/LuaAST.ts

```diff
@@ -791,15 +791,23 @@ export function createMethodCallExpression(
 export interface Identifier extends Expression {
     kind: SyntaxKind.Identifier;
     text: string;
+    symbol?: ts.Symbol;
 }
 
 export function isIdentifier(node: Node): node is Identifier {
     return node.kind === SyntaxKind.Identifier;
 }
 
-export function createIdentifier(text: string | ts.__String, tsOriginal?: ts.Node, parent?: Node): Identifier {
+export function createIdentifier(
+    text: string | ts.__String,
+    tsOriginal?: ts.Node,
+    symbol?: ts.Symbol,
+    parent?: Node
+): Identifier
+{
     const expression = createNode(SyntaxKind.Identifier, tsOriginal, parent) as Identifier;
     expression.text = text as string;
+    expression.symbol = symbol;
     return expression;
 }
```

Comment:
Again is it possible to avoid having ts types in the Lua AST?
Sorry I haven't elaborated on why I don't think TS stuff should be in an Lua AST.

Main reason, it's a design choice: Having no TS nodes in the AST clearly separates the transform and print step. Allowing them to run independent of each other. Not having any TS nodes in the AST allows us to easily create  Lua ASTs from the ground up (with no TS used to generate the AST).

I think a good compromise would be do just save the symbolID on the AST node, since that would just be a number and not a ts type. And as far as I can tell you only save the symbol to keep track on what should be hoisted? I think the ID could be enough for that purpose.
Or maybe map the tstl.Identifier to the Symbol somewhere else? For example in a Map<> in the Transformer, that way we wouldn't have to store the information on the node itself. Since you don't need that info in there print step there is no reason to save it on the node tbh.

Reply:
Ok, thanks for the explanation - that makes sense to me. All I need here is a way to differentiate unique symbols, so an ID system like you suggest should work.